src/sqlfluff/dialects/dialect_mysql.py

```diff
@@ -585,8 +597,13 @@ class IntervalExpressionSegment(BaseSegment):
     ParameterNameSegment=RegexParser(
         r"`?[A-Za-z0-9_]*`?", CodeSegment, name="parameter", type="parameter"
     ),
-    SingleIdentifierGrammar=ansi_dialect.get_grammar("SingleIdentifierGrammar").copy(
-        insert=[Ref("SessionVariableNameSegment")]
+    SingleIdentifierGrammar=Sequence(
+        Ref("SingleIdentifierSegment"),
+        Sequence(
+            Ref("AtSignLiteralSegment"),
+            Ref("SingleIdentifierSegment"),
+            optional=True,
+        ),
```

Comment:
I'd just define the full thing here, even though it's a bit of duplication:

```suggestion
    SingleIdentifierGrammar=Sequence(
        OneOf(
            Ref("NakedIdentifierSegment"),
            Ref("QuotedIdentifierSegment"),
            Ref("SessionVariableNameSegment"),
            Ref("SingleQuotedIdentifierSegment"),
            Ref("DoubleQuotedLiteralSegment"),
        ),
        Sequence(
            Ref("AtSignLiteralSegment"),
            OneOf(
                Ref("NakedIdentifierSegment"),
                Ref("QuotedIdentifierSegment"),
                Ref("SessionVariableNameSegment"),
                Ref("SingleQuotedIdentifierSegment"),
                Ref("DoubleQuotedLiteralSegment"),
            ),
            optional=True,
        ),
```

Reply:
Great, thanks for the suggestion!